### PR TITLE
1974 allow users to select which themes to export as variables

### LINKED ISSUE #1974: Allow users to select which themes to export as variables
**Allow users to select the themes to be created as variables.**
Currently all themes present are created as variables on clicking `Create Variables`

In #2406 we introduce a new flow for creating variables from the `Export to Figma` option

Image

- Allow users to select the themes that is to created as variables
- Only the selected themes groups will be created as Variables collection and the selected themes will be created as modes

This will allow the users to create local variables on different Figma files
#2023 will allow the references of variables from different Figma files to be maintained
-----------------
<details><summary>Original Issue</summary>
<p>
**Describe the bug**
I have 3 Figma files to manage Core, Semantic and Components tokens/variables. 
- Core file to manage Core tokens/variables,
- Semantic file to manage brands and themes related to them. 
- Components file to manage and publish the components library and hold tier 3 tokens (without publishing them).

Today if I open my Core file in Figma and then open Token Studio and enabling only Core sets. Then if I click on 'Create Variables' it created variables for all my tokens in TS, not only the one enabled...

**Expected behavior**
So what I want is to be able to export variables to Figma depending on themes enabled in Token Studio. Not everything, just themes enabled. 
And of course to be able to generate cross references through my Figma Files automatically (if my Core tokens are in a Core file and if in my Semantic file have references to tokens in my Core file I want a cross reference file token to be created).

It's important in term of maintenance because I can't imagine doing everything manually each time I add a token in Token Studio.

Cheers.
</p>
</details>

### PR DESCRIPTION
**Pull request by robinhoodie0823**
### Why does this PR exist?
The current implementation create all variables from all themes even user changed the selection of themes.

Closes #1974  

### What does this pull request do?
Updated the `handleExportToFigma` function to change when `selectedThemes` changed. And added a `createVariablesFromThemes` function to `useTokens` to create variables from selected themes.

### CODE REVIEW

File: packages/tokens-studio-for-figma/src/app/components/ManageStylesAndVariables/useExportThemesTab.tsx [hunk 1]
```diff
@@ -44,6 +44,7 @@ export default function useExportThemesTab() {
   const [selectedThemes, setSelectedThemes] = React.useState<string[]>(themes.map((theme) => theme.id));
 
   const handleSelectTheme = React.useCallback((themeId: string) => {
+    console.log('themeId: ', themeId);
```
Comment (LukeFinch) [inline packages/tokens-studio-for-figma/src/app/components/ManageStylesAndVariables/useExportThemesTab.tsx:47]: can remove

**Review by LukeFinch — APPROVED** [review]

**Review by six7 — CHANGES_REQUESTED** [review]
@robinhoodie0823 tests are failing here

File: packages/tokens-studio-for-figma/src/app/components/Settings/Settings.tsx [hunk 2]
```diff
@@ -79,22 +79,22 @@ function Settings() {
...
-      track('setPrefixStylesWithThemeName', { value: state });
+  // const handlePrefixWithThemeNameChange = React.useCallback(
+  //   (state: CheckedState) => {
+  //     track('setPrefixStylesWithThemeName', { value: state });
 
-      dispatch.settings.setPrefixStylesWithThemeName(!!state);
-    },
-    [dispatch.settings],
-  );
+  //     dispatch.settings.setPrefixStylesWithThemeName(!!state);
```
Comment (six7) [inline packages/tokens-studio-for-figma/src/app/components/Settings/Settings.tsx:94]: Why this comment?

File: packages/tokens-studio-for-figma/src/app/components/Settings/Settings.tsx [hunk 3]
```diff
@@ -79,22 +79,22 @@ function Settings() {
...
     getSessionId();
   });
-  const handleIgnoreChange = React.useCallback(
-    (state: CheckedState) => {
-      track('setIgnoreFirstPartForStyles', { value: state });
-      dispatch.settings.setIgnoreFirstPartForStyles(!!state);
-    },
-    [dispatch.settings],
-  );
+  // const handleIgnoreChange = React.useCallback(
```
Comment (six7) [inline packages/tokens-studio-for-figma/src/app/components/Settings/Settings.tsx:82]: Why is this a comment?

File: packages/tokens-studio-for-figma/src/app/components/Settings/Settings.tsx [hunk 4]
```diff
@@ -6,16 +6,16 @@ import { useTranslation } from 'react-i18next';
 import {
   Box, Link, Text, Button, Heading, Label, Stack, Switch,
 } from '@tokens-studio/ui';
-import ignoreFirstPartImage from '@/app/assets/hints/ignoreFirstPartForStyles.png';
-import prefixStylesImage from '@/app/assets/hints/prefixStyles.png';
+// import ignoreFirstPartImage from '@/app/assets/hints/ignoreFirstPartForStyles.png';
```
Comment (six7) [inline packages/tokens-studio-for-figma/src/app/components/Settings/Settings.tsx:9]: Lots of comments here

**Review by six7 — CHANGES_REQUESTED** [review]

**Review by six7 — APPROVED** [review]

**Comment by SamIam4Hyma** [comment]
Added to Featurebase Roadmap-	[❖❇️ Choose which themes to export as styles and variables](https://tokensstudio.featurebase.app/p/choose-which-themes-to-export-as-styles-and-variables)

**Pull request merged**